docs/rules/prefer-node-protocol.md

```diff
@@ -0,0 +1,52 @@
+# Enforce using the `node:` protocol when importing Node.js builtin modules (`n/prefer-node-protocol`)
+
+🔧 This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).
+
+<!-- end auto-generated rule header -->
+
+Older built-in Node modules such as fs now can be imported via either their name or `node:` + their name:
+
+```js
+import fs from "fs"
+import fs from "node:fs"
+```
+
+The prefixed versions are nice because they can't be overridden by user modules and are similarly formatted to prefix-only modules such as node:test.
+
+Note that Node.js support for this feature began in:
+
+> v16.0.0, v14.18.0 (`require()`)  
+> v14.13.1, v12.20.0 (`import`)
```

Comment:
We support setting the node version in the settings [docs/shared-settings.md#version](https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/shared-settings.md#version)

Would it be worth detecting and enabling/disabling this rule accordingly?

Reply:
I handled by https://github.com/eslint-community/eslint-plugin-n/pull/183/commits/dc1bcbc252d11830f14258094bef3862d28531fd
Does it match what you are thinking?

Reply:
This is exactly what I was thinking!